### PR TITLE
Added C++17 support

### PR DESCRIPTION
**Pull request by mguludag**
Hi, I added C++17 support and cpp version check macros 👍

### CODE REVIEW

File: Invoker.hpp [hunk 1]
```diff
@@ -1,25 +1,45 @@
+#ifndef INVOKER_HEADER
```
Comment (calebxyz) [inline Invoker.hpp]: First of all thank you.

please use #pragma once

File: Invoker.hpp [hunk 2]
```diff
@@ -1,25 +1,45 @@
...
-    std::add_pointer_t<Invocable>, Invocable>;
-    template <typename I> 
+    using Invocable_T = std::conditional_t<std::is_function_v<Invocable>, std::add_pointer_t<Invocable>, Invocable>;
+#if __cplusplus > 201703L
+    template <typename I>
     requires std::convertible_to<I, Invocable_T>
-    Invoker(I invocable): 
-    invocable_(std::move(invocable)) {} 
+#else // __cplusplus < 202002L
+    template <typename I, typename = typename std::enable_if<std::is_convertible<I, Invocable_T>::value>::type>
```
Comment (calebxyz) [inline Invoker.hpp]: please change this to:
template <typename I, std::enable_if_t<std::is_convertible_v<I, Invocable_T>>>

File: Invoker.hpp [hunk 3]
```diff
@@ -1,25 +1,45 @@
...
+#endif // __cplusplus > 201703L
+    Invoker(I invocable) : invocable_(std::move(invocable)) {}
     
+#if __cplusplus > 201703L
     template <typename... Args>
     requires std::invocable<Invocable&, Args...>
+#else
+    template <typename... Args, typename = typename std::enable_if<std::is_invocable<Invocable_T, Args...>::value>::type>
```
Comment (calebxyz) [inline Invoker.hpp]: please use: 
template <typename... Args, std::enable_if_t<std::is_invocable_v<Invocable_T, Args...>>>

Comment (calebxyz) [reply Invoker.hpp]: actually in both enable_if cases i would use 
static_assert instead in the body of the code 
that way the error messages will be much clearer , enable_if tend to produce a lot of noise

**Review by calebxyz — CHANGES_REQUESTED** [review]

File: Invoker.hpp (hunk 3 above)
Comment (mguludag) [reply Invoker.hpp]: in [b6787a5](https://github.com/mguludag/Invoker/commit/b6787a5326f4635dc791a730d0734c31874eaaa8) changed `enable_if `to `static_assert `and include guards to `#pragma once`

**Review by calebxyz — APPROVED** [review]

**Pull request merged**